bftengine/include/bftengine/ControlStateManager.hpp

```diff
@@ -14,19 +14,47 @@
 #pragma once
 #include <optional>
 #include "IStateTransfer.hpp"
+#include "ReservedPages.hpp"
+#include "Serializable.h"
 
 namespace bftEngine {
 
-class ControlStateManager {
+class ControlStateManager : public ResPagesClient<ControlStateManager, 1, 1> {
  public:
   void setStopAtNextCheckpoint();
-  std::optional<uint64_t> getStopCheckpointToStopAt();
+  std::optional<bool> getStopCheckpointToStopAt();
 
   ControlStateManager(IStateTransfer& state_transfer);
   ControlStateManager& operator=(const ControlStateManager&) = delete;
   ControlStateManager(const ControlStateManager&) = delete;
+  ~ControlStateManager() {}
 
  private:
   IStateTransfer& state_transfer_;
+
+  static constexpr uint32_t size_of_reserved_pages_ = 512;
```

Comment:
Size of reserved page is defined in configuration and can't be changed by a module

Reply:
The size is unique for all the reserved pages users?

Reply:
Not directly related to this PR, but to the resPagesClients themselves. What happens if the number of reserved pages used by all clients exceeds the configuration? Do we have any detection of this? @toly-kournik

Reply:
fixed